c/cmake_modules/AdbcDefines.cmake

```diff
@@ -92,6 +92,17 @@ elseif(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang"
       -Werror
       -Wno-unused-parameter)
   set(ADBC_C_CXX_FLAGS_PRODUCTION -Wall)
+
+  string(APPEND C_DEBUG_FLAGS " -Og -ggdb")
+  string(APPEND CXX_DEBUG_FLAGS " -Og -ggdb")
+  string(APPEND C_RELWITHDEBINFO_FLAGS " -Og -ggdb")
+  string(APPEND CXX_RELWITHDEBINFO_FLAGS " -Og -ggdb")
+
+  if("${_lower_build_type}" STREQUAL "debug" AND "${CMAKE_CXX_FLAGS}" MATCHES "-O2")
+    # Activate scripts for Conda compilers set CXXFLAGS etc. with -O2.
+    message(WARNING "This is a debug build, but your CXXFLAGS have -O2. This is common in Conda environments.\nReset your CFLAGS, CPPFLAGS, and CXXFLAGS to -Og for debugging."
```

Comment:
FYI: Users can use `-DCMAKE_C_FLAGS_DEBUG=-Og -DCMAKE_CXX_FLAGS_DEBUG=-Og` instead of resetting `CFLAGS`/`CXXFLAGS`.

Here are my related tries on apache/arrow:
* https://github.com/apache/arrow/pull/15022
* https://github.com/apache/arrow/pull/35924

Reply:
I'll give this a shot, thanks!

Reply:
> If you use multiple -O options, with or without level numbers, the last such option is the one that is effective. 

Ah, I think this is what I was not understanding earlier.